File: 2021-09/숫자문자열과영단어.js
function solution(s) {
    const numbers = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine'];

    for(let i = 0; i < 10; i++) {
        while(s.match(numbers[i])) {
            s = s.replace(numbers[i], i);
        }
    }

    return +s;
}

console.log(solution("one4seveneight")); //1478
console.log(solution("23four5six7")); //234567
console.log(solution("2three45sixseven")); //234567
console.log(solution("123")); //123


/* 다른 사람이 짠 배우고 싶은 코드 */
function solution2(s) {
    s = s.replace(/zero/gi, 0)
    .replace(/one/gi, 1)
    .replace(/two/gi, 2)
    .replace(/three/gi, 3)
    .replace(/four/gi, 4)
    .replace(/five/gi, 5)
    .replace(/six/gi, 6)
    .replace(/seven/gi, 7)
    .replace(/eight/gi, 8)
    .replace(/nine/gi, 9)
    return parseInt(s);
}

File: programmers/src/main/java/level1/kakao/숫자문자열과영단어.java
package level1.kakao;

public class 숫자문자열과영단어 {

	private static final String[] numbers = {"zero", "one", "two", "three",
		"four", "five", "six", "seven",
		"eight", "nine"};

	public int solution(String s) {
		for (int i = 0; i < 10; i++) {
			s = s.replaceAll(numbers[i], Integer.toString(i));
		}
		return Integer.parseInt(s);
	}

}
